### ISSUE TITLE
PerformanceProfiler only instrumented on CUDA dispatch path; CPU MatMul kernels emit no timings even with Instance.Enabled=true

### ISSUE THREAD

**Issue by ooples** [ooples/AiDotNet.Tensors#363]
## Summary

`AiDotNet.Tensors.Engines.Optimization.PerformanceProfiler.Instance` is the documented hook for per-kernel timing rollups (consumed by `AiDotNet.Diagnostics.TensorsOperationProfile.Capture`). Setting `PerformanceProfiler.Instance.Enabled = true` and then running CPU-side inference produces **zero** recorded operations.

Sole call site is the CUDA dispatch path:

```
src/AiDotNet.Tensors/Engines/DirectGpu/CUDA/CudaDispatchPolicy.cs:102
    return PerformanceProfiler.Instance.Profile(label);
```

CPU-side `IEngine.TensorMatMul`, softmax, dense, embedding, attention, etc. never invoke `PerformanceProfiler.Instance.Profile(...)`, so CPU forward passes leave the profiler empty even when `Enabled` is on.

## Repro

HarmonicEngine `AiDotNetProfilingTransformerLMPredictor` + `Phase_PAPER_A_PathB_Profiling_Sanity` test on CPU:

```csharp
PerformanceProfiler.Instance.Clear();
PerformanceProfiler.Instance.Enabled = true;
// ... drive ~300 _trained.Predict(input) calls (Transformer<float>, dModel=128, L=2, ctx=64) ...
var profile = TensorsOperationProfile.Capture();
// profile.Operations.Count == 0
// profile.TotalMilliseconds == 0
```

## Expected

CPU MatMul and other hot tensor kernels surface call counts + timings (or at minimum the BLAS-routed GEMM call) so the documented \"PyTorch-parity per-op breakdown\" claim on `TensorsOperationProfile` (see AiDotNet `AiModelResult.Diagnostics.cs` docstring referencing \"low-level `torch.profiler.profile` CUDA/CPU op breakdown\") actually returns data on CPU.

## Impact

`AiModelResult.TensorsOperationProfile` is unusable for CPU diagnostics. Anyone trying to find CPU bottlenecks via the documented API gets a misleadingly empty report and has to guess whether their code is slow or just not being measured.

## Suggested fix

Either:
1. Add `PerformanceProfiler.Instance.Profile(\"matmul\")` (and friends) around CPU GEMM / softmax / layernorm dispatch entry points in `Engines/Cpu/*` (or wherever the CPU IEngine routes ops).
2. Document the CUDA-only scope explicitly on `PerformanceProfiler` and `TensorsOperationProfile.Capture` so users know not to expect CPU coverage.

Option 1 is the user-facing fix. Option 2 unblocks the workaround at minimum.

## Observed alongside

HarmonicEngine `AiDotNetProfilingTransformerLMPredictor.cs` + `Phase_PAPER_A_PathB_Profiling_Sanity.cs` on branch `research/ledger-overhaul-and-l4-reification`. P50/P95/P99 from the higher-level `AiDotNet.Diagnostics.ProfilerSession` work fine (P50=34.751ms, P95=48.195ms, P99=58.141ms on dModel=128/L=2/ctx=64/300 positions); only the Tensors-level kernel rollup is silently empty.
